packages/react/slider/src/Slider.tsx

```diff
@@ -493,12 +497,18 @@ type SliderThumbPrimitive = Polymorphic.ForwardRefComponent<
 >;
 
 const SliderThumb = React.forwardRef((props, forwardedRef) => {
-  const { ref: collectionRef, index } = useSliderCollectionItem();
-  const ref = useComposedRefs(forwardedRef, collectionRef);
   const context = useSliderContext(THUMB_NAME);
+  const { getItems } = useCollectionItem();
+  const [thumb, setThumb] = React.useState<React.ElementRef<typeof SliderThumbImpl> | null>(null);
+  const composedRefs = useComposedRefs(forwardedRef, (node) => setThumb(node));
+  const index = React.useMemo(
+    () => (thumb ? getItems().findIndex((item) => item.ref.current === thumb) : 0),
```

Comment:
Curious, could `value` technically be inaccurate on the first render if we're falling back to `0` here?

Reply:
Yep, that's the whole SSR index issue I was explaining. The old collection stuff tried to solve this.

Reply:
This is challenging my understanding as well haha